juicefs-1.3.yaml

```diff
@@ -0,0 +1,133 @@
+  copyright:
+    - license: Apache-2.0
+  dependencies:
+    runtime:
+      - fuse3
+    provides:
+      - juicefs=${{package.full-version}}
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/juicedata/juicefs
+      tag: v${{package.version}}
+      expected-commit: 30190ca1094d26e85f19a979ca51b0ea19af1eaa
+
+  - uses: go/bump
+    with:
+      deps: |-
+        golang.org/x/net@v0.38.0
+        github.com/golang/glog@v1.2.4
+        github.com/tidwall/gjson@v1.9.3
+        golang.org/x/oauth2@v0.27.0
+
+  - uses: go/build
+    with:
+      packages: .
+      output: juicefs
+      ldflags: |
+        -X github.com/juicedata/juicefs/pkg/version.version=${{package.version}}
+        -X github.com/juicedata/juicefs/pkg/version.revision=$(git rev-parse HEAD)
+        -X github.com/juicedata/juicefs/pkg/version.revisionDate=$(date -u -d "@$SOURCE_DATE_EPOCH" +"%Y-%m-%dT%H:%M:%SZ")
+
+subpackages:
+  - name: ${{package.name}}-compat
+    description: Compat package for ${{package.name}}
+    dependencies:
+      runtime:
+        - lsof
+        - procps
+        - strace
+        - iproute2
+        - net-tools
+      provides:
+        - juicefs-compat=${{package.full-version}}
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.contextdir}}"/usr/local/bin
+          mkdir -p "${{targets.contextdir}}"/bin
+          ln -sf /usr/bin/juicefs "${{targets.contextdir}}"/usr/local/bin/juicefs
+          ln -sf /usr/bin/juicefs "${{targets.contextdir}}"/bin/mount.juicefs
+    test:
+      pipeline:
+        - runs: |
+            test "$(readlink /usr/local/bin/juicefs)" = "/usr/bin/juicefs"
+            test "$(readlink /bin/mount.juicefs)" = "/usr/bin/juicefs"
+
+update:
+  enabled: true
+  ignore-regex-patterns:
+    - '-rc'
+    - '-beta'
+    - '-dev'
+  github:
+    identifier: juicedata/juicefs
+    strip-prefix: v
+    tag-filter: v1.3
+
+test:
+  environment:
+    contents:
+      packages:
+        - redis
+        - minio
+        - mc
+        - wait-for-it
+        - curl
+  pipeline:
+    - name: "Version test"
+      runs: |
+        juicefs --version 2>&1 | grep ${{package.version}}
+    - name: "Start redis for metadata"
+      runs: |
+        redis-server --daemonize yes > /tmp/redis.logs 2>&1 &
+        sleep 2
+    - name: "Start MinIO"
+      runs: |
+        export MINIO_ACCESS_KEY=minioadmin
+        export MINIO_SECRET_KEY=minioadmin
+        mkdir -p /data/minio
+        minio server /data/minio --address :9000 --console-address :9001 > /tmp/minio.log 2>&1 &
+        wait-for-it localhost:9000 -t 30
+    - name: "Create test bucket using mc"
+      runs: |
+        export MINIO_ACCESS_KEY=minioadmin
+        export MINIO_SECRET_KEY=minioadmin
+        mc alias set local http://localhost:9000 minioadmin minioadmin > /tmp/mc.log 2>&1
+        mc mb local/juicefs-bucket
+        sleep 2 # Wait for juicefs bucket to show up
```

Comment:
Can we add a retry on failure? This way we'd avoid unnecessary backpressure in CI